deployment/src/strongmind_deployment/autoscale.py

```diff
@@ -0,0 +1,202 @@
+        :key worker_min_number_of_instances: The minimum number of instances available in the scaling policy for the worker.
+        :key worker_autoscale_threshold: The threshold for the worker autoscaling policy. Default is 3.
+        """
+        super().__init__('strongmind:global_build:commons:worker-autoscale', name, None, opts)
+        self.project_stack = pulumi.get_project() + "-" + pulumi.get_stack()
+        self.worker_max_capacity = kwargs.get('worker_max_number_of_instances', 1)
+        self.worker_min_capacity = kwargs.get('worker_min_number_of_instances', 1)
+        self.threshold = kwargs.get('worker_autoscale_threshold', 3)
+        self.autoscaling()
+
+# scale out based on EnqueuedJobs metric
+    def autoscaling(self):
+        self.worker_autoscaling_target = aws.appautoscaling.Target(
+            "worker-autoscaling_target",
+            max_capacity=self.worker_max_capacity,
+            min_capacity=self.worker_min_capacity,
+            resource_id=f"service/{self.project_stack}/{self.project_stack}-worker",
+            scalable_dimension="ecs:service:DesiredCount",
+            service_namespace="ecs",
+        )
+        self.worker_autoscaling_out_policy = aws.appautoscaling.Policy(
+            "worker_autoscaling_out_policy",
+            name=f"{self.project_stack}-worker-autoscaling-out-policy",
+            policy_type="StepScaling",
+            resource_id=self.worker_autoscaling_target.resource_id,
+            scalable_dimension=self.worker_autoscaling_target.scalable_dimension,
+            service_namespace=self.worker_autoscaling_target.service_namespace,
+            step_scaling_policy_configuration=aws.appautoscaling.PolicyStepScalingPolicyConfigurationArgs(
+                adjustment_type="ChangeInCapacity",
+                cooldown=60,
+                metric_aggregation_type="Maximum",
+                step_adjustments=[
+                    aws.appautoscaling.PolicyStepScalingPolicyConfigurationStepAdjustmentArgs(
+                        metric_interval_upper_bound="10",
+                        metric_interval_lower_bound="0",
+                        scaling_adjustment=1,
+                    ),
+                    aws.appautoscaling.PolicyStepScalingPolicyConfigurationStepAdjustmentArgs(
+                        metric_interval_lower_bound="10",
+                        scaling_adjustment=3,
+                    )
+                ],
+            )
+        )
+        self.worker_autoscaling_out_alarm = aws.cloudwatch.MetricAlarm(
+            "worker_autoscaling_out_alarm",
+            name=f"{self.project_stack}-worker-auto-scaling-out-alarm",
+            comparison_operator="GreaterThanThreshold",
+            evaluation_periods=1,
+            metric_name="EnqueuedJobs",
+            unit="Count",
+# DIMENSIONS ARE COMMENTED HERE until we can update the sidekiqcloudwatchmetrcis gem to accept dimeensions
```

Comment:
Does this PR need to wait on a change to sidekiqcloudwatchmetrics gem?

Reply:
So long as no one tries to put it in place by using the worker_autoscale = True it does not matter who goes first.

Reply:
No, we are not publishing dimensions with the enqueued jobs metric.